state/bakerystorage/config.go

```diff
@@ -0,0 +1,169 @@
+	if err != nil {
+		return txn.Op{}, errors.Trace(err)
+	}
+	localUsersThirdPartyKeyBytes, err := json.Marshal(localUsersThirdPartyKey)
+	if err != nil {
+		return txn.Op{}, errors.Trace(err)
+	}
+
+	externalUsersThirdPartyKey, err := bakery.GenerateKey()
+	if err != nil {
+		return txn.Op{}, errors.Trace(err)
+	}
+	externalUsersThirdPartyKeyBytes, err := json.Marshal(externalUsersThirdPartyKey)
+	if err != nil {
+		return txn.Op{}, errors.Trace(err)
+	}
+
+	localUsersKey, err := bakery.GenerateKey()
+	if err != nil {
+		return txn.Op{}, errors.Trace(err)
+	}
+	localUsersKeyBytes, err := json.Marshal(localUsersKey)
+	if err != nil {
+		return txn.Op{}, errors.Trace(err)
+	}
+
+	offersThirdPartyKey, err := bakery.GenerateKey()
+	if err != nil {
+		return txn.Op{}, errors.Trace(err)
+	}
+	offersThirdPartyKeyBytes, err := json.Marshal(offersThirdPartyKey)
+	if err != nil {
+		return txn.Op{}, errors.Trace(err)
+	}
+
+	return txn.Op{
+		C:      b.collection,
+		Id:     bakeryConfigKey,
+		Assert: txn.DocMissing,
+		Insert: &bakeryConfigDoc{
+			LocalUsersKey:              string(localUsersKeyBytes),
+			LocalUsersThirdPartyKey:    string(localUsersThirdPartyKeyBytes),
+			ExternalUsersThirdPartyKey: string(externalUsersThirdPartyKeyBytes),
+			OffersThirdPartyKey:        string(offersThirdPartyKeyBytes),
+		},
+	}, nil
+}
+
+// GetLocalUsersKey returns the key pair used with the local users bakery.
+func (b *bakeryConfig) GetLocalUsersKey() (*bakery.KeyPair, error) {
+	doc, err := b.bakeryConfigDoc()
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	return b.deserialiseKey(doc.LocalUsersKey, "local users key")
+}
+
+// GetLocalUsersThirdPartyKey returns the third party key pair used with the local users bakery.
+func (b *bakeryConfig) GetLocalUsersThirdPartyKey() (*bakery.KeyPair, error) {
+	doc, err := b.bakeryConfigDoc()
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	return b.deserialiseKey(doc.LocalUsersThirdPartyKey, "local users third party key")
+}
+
+// GetExternalUsersThirdPartyKey returns the third party key pair used with the external users bakery.
+func (b *bakeryConfig) GetExternalUsersThirdPartyKey() (*bakery.KeyPair, error) {
+	doc, err := b.bakeryConfigDoc()
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	return b.deserialiseKey(doc.ExternalUsersThirdPartyKey, "external users third party key")
+}
+
+// GetOffersThirdPartyKey returns the key pair used with the cross model offers bakery.
+func (b *bakeryConfig) GetOffersThirdPartyKey() (*bakery.KeyPair, error) {
+	doc, err := b.bakeryConfigDoc()
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	return b.deserialiseKey(doc.OffersThirdPartyKey, "offers third party key")
+}
+
+func (b *bakeryConfig) deserialiseKey(data, label string) (*bakery.KeyPair, error) {
+	if data == "" {
+		return nil, errors.NotValidf("empty " + label)
+	}
+
+	var keyPair bakery.KeyPair
+	err := json.Unmarshal([]byte(data), &keyPair)
+	if err != nil {
+		err = errors.Trace(err)
```

Comment:
I've made a PR to fix this https://github.com/juju/juju/pull/11733/files#diff-7f05070aef97e2fdd5573b2bc8fcdc99R151